src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BitSwapApiTest.cs

```diff
@@ -0,0 +1,528 @@
+        //             var otherOriginalProtocols = (_dfsServiceOther.BitSwapService).Protocols;
+        //
+        //             (_dfsService.BitSwapService).Protocols = new IBitswapProtocol[]
+        //             {
+        //                 new Bitswap1
+        //                 {
+        //                     BitswapService = _dfsService.BitSwapService
+        //                 }
+        //             };
+        //             
+        //             _dfsService.Options.Discovery.DisableMdns = true;
+        //             _dfsService.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+        //             await _dfsService.StartAsync();
+        //
+        //             (_dfsServiceOther.BitSwapService).Protocols = new IBitswapProtocol[]
+        //             {
+        //                 new Bitswap1
+        //                 {
+        //                     BitswapService = _dfsServiceOther.BitSwapService
+        //                 }
+        //             };
+        //             
+        //             _dfsServiceOther.Options.Discovery.DisableMdns = true;
+        //             _dfsServiceOther.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+        //             await _dfsServiceOther.StartAsync();
+        //             
+        //             try
+        //             {
+        //                 var data = Guid.NewGuid().ToByteArray();
+        //                 var cid = await _dfsServiceOther.BlockApi.PutAsync(data);
+        //
+        //                 var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+        //                 var getTask = _dfsService.BlockApi.GetAsync(cid, cts.Token);
+        //
+        //                 var remote = _dfsServiceOther.LocalPeer;
+        //                 await _dfsService.SwarmApi.ConnectAsync(remote.Addresses.First(), cts.Token);
+        //                 var block = await getTask;
+        //
+        //                 Assert.False(getTask.IsCanceled, "task cancelled");
+        //                 Assert.False(getTask.IsFaulted, "task faulted");
+        //                 Assert.True(getTask.IsCompleted, "task not completed");
+        //                 Assert.Equal(cid, block.Id);
+        //                 Assert.Equal(data, block.DataBytes);
+        //
+        //                 var otherPeer = _dfsServiceOther.LocalPeer;
+        //                 var ledger = await _dfsService.BitSwapApi.LedgerAsync(otherPeer, cts.Token);
+        //                 Assert.Equal(otherPeer, ledger.Peer);
+        //                 Assert.NotEqual(0UL, ledger.BlocksExchanged);
+        //                 Assert.NotEqual(0UL, ledger.DataReceived);
+        //                 Assert.Equal(0UL, ledger.DataSent);
+        //                 Assert.True(ledger.IsInDebt);
+        //
+        //                 // TODO: Timing issue here.  ipfsOther could have sent the block
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)